app/src/main/java/com/example/b2blearning/MainActivity.java

```diff
@@ -5,22 +5,65 @@
 import android.os.Bundle;
 import android.view.View;
 import android.widget.Button;
+import android.widget.EditText;
+import android.widget.Toast;
+
+import retrofit2.Call;
+import retrofit2.Callback;
+import retrofit2.Response;
+
 
 public class MainActivity extends AppCompatActivity {
+    private EditText emailView;
+    private EditText passwordView;
+    private boolean check;
 
     @Override
     protected void onCreate(Bundle savedInstanceState) {
         super.onCreate(savedInstanceState);
         setContentView(R.layout.activity_main);
 
+        emailView = findViewById(R.id.login);
+        passwordView = findViewById(R.id.password);
+
+
+        //Instantiating login button.
         Button login = (Button)findViewById(R.id.loginBtn);
 
+        //Enabling button click and it's functionality.
         login.setOnClickListener(new View.OnClickListener() {
             @Override
             public void onClick(View v) {
+                iLoginConfig login = iLoginConfig.retrofit.create(iLoginConfig.class);
+
+                String email = emailView.getText().toString();
+                String password = passwordView.getText().toString();
+                check = Utility.loginCheck(email, password, getBaseContext());
+
+                if(check){
+                    login.directions(email, password).enqueue(new Callback<User>() {
+                        @Override
+                        public void onResponse(Call<User> call, Response<User> response) {
+                            int code = response.code();
+                            if (code == 200) {
+                                Toast.makeText(getBaseContext(), "Login response " + code + ": Successful", Toast.LENGTH_LONG).show();
+                            } else {
+                                Toast.makeText(getBaseContext(), "Login response : Error " + code + ", incorrect credentials", Toast.LENGTH_LONG).show();
```

Comment:
Verify if all other codes returned are related to "incorrect credentials".
If not, divide the answers in another possibilities.
@LuisFernandoSFonseca

Reply:
The only error code the server return is 401(Unauthorized)